cmd/main_test.go

```diff
@@ -168,8 +174,201 @@ func Test_assignAddress(t *testing.T) {
+			}
+		})
+	}
+}
+
+func Test_waitForAddressToBeReported(t *testing.T) {
+	type args struct {
+		c          context.Context
+		explorerFn func(t *testing.T) node.Explorer
+		node       *types.Node
+		address    string
+		cfg        *config.Config
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "address reported with no retries",
+			args: args{
+				c:       context.Background(),
+				address: "1.1.1.1",
+				explorerFn: func(t *testing.T) node.Explorer {
+					mock := nodeMocks.NewExplorer(t)
+					mock.EXPECT().GetNode(tmock.Anything, "test-node").Return(
+						&types.Node{
+							Name:        "test-node",
+							Instance:    "test-instance",
+							Region:      "test-region",
+							Zone:        "test-zone",
+							ExternalIPs: []net.IP{net.IPv4(1, 1, 1, 1)},
+						},
+						nil,
+					)
+					return mock
+				},
+				node: &types.Node{
+					Name:     "test-node",
+					Instance: "test-instance",
+					Region:   "test-region",
+					Zone:     "test-zone",
+				},
+				cfg: &config.Config{
+					Filter:        []string{"test-filter"},
+					OrderBy:       "test-order-by",
+					RetryAttempts: 3,
+					RetryInterval: time.Millisecond,
+					LeaseDuration: 1,
+				},
+			},
+		},
+		{
+			name: "address reported after a few retries",
+			args: args{
+				c:       context.Background(),
+				address: "1.1.1.1",
+				explorerFn: func(t *testing.T) node.Explorer {
+					mock := nodeMocks.NewExplorer(t)
+					mock.EXPECT().GetNode(tmock.Anything, "test-node").Return(&types.Node{
+						Name:        "test-node",
+						Instance:    "test-instance",
+						Region:      "test-region",
+						Zone:        "test-zone",
+						ExternalIPs: []net.IP{net.IPv4(9, 9, 9, 9)},
+					}, nil).Times(3)
+					mock.EXPECT().GetNode(tmock.Anything, "test-node").Return(&types.Node{
+						Name:        "test-node",
+						Instance:    "test-instance",
+						Region:      "test-region",
+						Zone:        "test-zone",
+						ExternalIPs: []net.IP{net.IPv4(1, 1, 1, 1)},
+					}, nil).Once()
+					return mock
+				},
+				node: &types.Node{
+					Name:     "test-node",
+					Instance: "test-instance",
+					Region:   "test-region",
+					Zone:     "test-zone",
+				},
+				cfg: &config.Config{
+					Filter:        []string{"test-filter"},
+					OrderBy:       "test-order-by",
+					RetryAttempts: 3,
+					RetryInterval: time.Millisecond,
+					LeaseDuration: 1,
+				},
+			},
+		},
+		{
+			name: "error after a few retries and reached maximum number of retries",
+			args: args{
+				c: context.Background(),
+				explorerFn: func(t *testing.T) node.Explorer {
+					mock := nodeMocks.NewExplorer(t)
+					mock.EXPECT().GetNode(tmock.Anything, "test-node").Return(&types.Node{
+						Name:        "test-node",
+						Instance:    "test-instance",
+						Region:      "test-region",
+						Zone:        "test-zone",
+						ExternalIPs: []net.IP{net.IPv4(9, 9, 9, 9)},
+					}, nil).Times(4)
+					return mock
+				},
```

Comment:
Worth adding to return the correct IP after the retries, to verify we still get an error because we gave up too early?